store/store.go

```diff
@@ -309,20 +326,25 @@ func (s *Store[H]) Append(ctx context.Context, headers ...H) error {
 	var err error
 	// take current write head to verify headers against
 	var head H
-	headPtr := s.writeHead.Load()
-	if headPtr == nil {
+	if headPtr := s.writeHead.Load(); headPtr == nil {
 		head, err = s.Head(ctx)
 		if err != nil {
 			return err
 		}
+		// store header from the disk.
+		gotHead := head
+		s.writeHead.CompareAndSwap(nil, &gotHead)
```

Comment:
why not just directly store instead of using CAS here? IIUC the only other place writeHead is written to is also called from `Append` after this call here and `Append` is blocking until all writes finish.

Reply:
Theoretically we might have 2 concurrent `.Append` calls. Both with non-initialized `s.writeHead`. Instead of blindly overwriting `writeHead` only 1 goroutine will win.

Probably this is not a big deal what header will be written, will change to `s.write.Head.Store(...)`.